c-example/README.md

```diff
@@ -0,0 +1,63 @@
+# C example
+
+This is the README file for spglib examples for C.
+
+## Examples of using spglib in a cmake project
+
+See the local [`CMakeLists.txt`](CMakeLists.txt) file for how to use
+spglib in a cmake project, using either native cmake options or pkg-config.
+
+## Examples of linking to spglib manually
+
+To manually compile `example.c`:
+
+```console
+gcc example.c -I[include directory] -L[library directory] -lsymspg -lm
+```
+
+where the `[include directory]` and `[library directory]` can be obtained
+via `pkg-config --cflags --libs spglib`.
+
+## Concept
+
+A crystal structure is given to a spglib function to obtain symmetry
+information. `spg_get_dataset` is the function that returns most fruitful
+space group information. This returns the result in the C-structure
+`SpglibDataset`. We don't know the length of symmetry operations to be returned,
+the memory spaces are allocated dynamically, which means we have to free them.
```

Comment:
That's what you meant with allocatable in Fortran. I think this needs to be tested in an IDE to confirm this behavior.

Note that using SWIG, we can share C++ smart pointers such that the other language can decide how to deallocate the memory. It has both Python and Fortran bindings.

Reply:
In the Fortran wrapper:

https://github.com/spglib/spglib/blob/02159eef6e7349535049a43fe2272bb634c77945/fortran/spglib_f08.f90#L525

the data in `SpglibDataset` in C structure is copied to the Fortran structure in which the data are allocated by Fortran's allocatable arrays, which should be deallocated automatically as the language specification.

I have no idea about SWIG. For me, writing Python C-API is easy. But for long term perspective, if maintainers can prefer SWIG or any other utils (e.g. cython in python), we can move. Until now, I could be only the person who maintains spglib in long-run, so it has been safer to choose the ways that are easiest for me. However this way is clearly not sustainable although it seems spglib is expected to live longer.

Reply:
> the data in `SpglibDataset` in C structure is copied to the Fortran structure

But that means the original dataset generated by C still needs to be deleted.

> I have no idea about SWIG. For me, writing Python C-API is easy.

Basically SWIG = you don't need to write the binding api, it generates for you. You just point it to the relevant functions/classes and it generates the bindings necessary (including C-Fortran string passing, oh the horror in doing that manually)

Reply:
> But that means the original dataset generated by C still needs to be deleted.

This is done inside the Fortran wrapper.

https://github.com/spglib/spglib/blob/02159eef6e7349535049a43fe2272bb634c77945/fortran/spglib_f08.f90#L669

Reply:
👍 Whenever I look at C and Fortran code I miss the automatic memory management of C++